CAN-135: Manage Profile: Implement Google Places API for address

vikasranatalentelgia opened the pull request:
Manage Profile: Implement Google Places API for address

vikasranatalentelgia committed:
CAN-135: Manage Profile: Implement Google Places API for address

Manage Profile: Implement Google Places API for address

vikasranatalentelgia committed:
CAN-135: Move Google API Key to Environment Variable

env variable: NEXT_PUBLIC_GOOGLE_API_KEY

vikasranatalentelgia committed:
Google API key variable added to .env.example

Google API key env variable added to .env.example

vikasranatalentelgia committed:
Updated .env.example files with dummy API key

Updated .env.example files with dummy API key

akuks commented on src/components/ComponentPages/ProfileInfo/index.tsx:
```diff
@@ -124,6 +128,65 @@ const ProfileInfo = () => {
+      place.address_components.find(c => c.types.includes('postal_code')) || {};
+    var city = ""
+    var country = ""
+    var address2 = ""
```
How about this
` let city = country = address2 = ""`

vikasranatalentelgia replied:
done

akuks commented on src/components/ComponentPages/ProfileInfo/index.tsx:
```diff
@@ -124,6 +128,65 @@ const ProfileInfo = () => {
+    form.setFieldsValue({ ["city"]: city })
+    form.setFieldsValue({ ["state"]: state })
+    form.setFieldsValue({ ["country"]: country })
+  }
```
Probably we need to break this. This doesn't looks good

```javascript

for (const component of results[0].address_components) {
      const componentType = component.types[0];
      
      address2 = getAddress(componentType, address2, component);

 .....
......
.....
}


// Create a function similar to this
function getAddress( type, address, component ) {
    if ( type === "neighbourhood" ) {
        return component.long_name
    }
    else if ( type.match( /^neighborhood$|^sublocality_level_2$|^sublocality_level_1$/ ) ) {
        return address + ", " + component.long_name
    }
    else { 
        return 0
    }
  }

```

akuks replied:
`city`, `state` and `country` are state forward

vikasranatalentelgia replied:
done

akuks reviewed (changes requested):
Please go through the comments

vikasranatalentelgia committed:
Merge branch 'development' into CAN-135

vikasranatalentelgia committed:
CAN-135: Manage Profile: Implement Google Places API for address  Code refactoring

CAN-135: Manage Profile: Implement Google Places API for address Code refactoring
